HW0/P5/P5.txt

```diff
@@ -1 +1,10 @@
-# Your answers here.
+1. 255 seconds.
+2. Yes. There are 8 people totally. They can distribute 32 bags to one person and each person only count their own bags. 31+3=34 seconds. In this case, they just need 34 seconds to do it.
+3. If there are infinite number of people. They can use divide and conquer algorithm. They can use 128 people to sum each 2 bags first. And then sum them two by two at the same time and continue again until we have one sum. This strategy takes log(256)=8 seconds.
+4. If we have N bags. The fastest time to count them would be ceil(log2(N))
```

Comment:
I agree with this answer, but it would have been nice to have a little more discussion on why ceil is needed.